pkg/k8s/service_cache.go

```diff
@@ -369,3 +377,100 @@ func (s *ServiceCache) UniqueServiceFrontends() map[string]struct{} {
 
 	return uniqueFrontends
 }
+
+// correlateEndpoints builds a combined Endpoints of the local endpoints and
+// all external endpoints if the service is marked as a global service
+func (s *ServiceCache) correlateEndpoints(id ServiceID) *Endpoints {
+	endpoints := newEndpoints()
+
+	localEndpoints, ok := s.endpoints[id]
+	if ok {
+		for ip, e := range localEndpoints.Backends {
+			endpoints.Backends[ip] = e
+		}
+	}
+
+	svc, ok := s.services[id]
+	if ok && svc.IncludeExternal {
```

Comment:
Invert the if to avoid nested branches.

Reply:
You seem to like this. I don't. What is bad about nesting? Returning early makes it a lot harder to understand the entire function as every single exit point must be understood. I don't follow why this is better.